x-pack/plugins/enterprise_search/server/lib/get_oauth_token_package_params.test.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ENTERPRISE_SEARCH_KIBANA_COOKIE } from '../../common/constants';
+
+import { getOAuthTokenPackageParams } from './get_oauth_token_package_params';
+
+describe('getOAuthTokenPackage', () => {
+  const tokenPackage = 'sEkReTZ';
+  const tokenPackageCookie = `${ENTERPRISE_SEARCH_KIBANA_COOKIE}=${tokenPackage}`;
+  const tokenPackageParams = { token_package: tokenPackage };
+
+  describe('when there are no cookie headers', () => {
+    it('returns an empty parameter set', () => {
+      expect(getOAuthTokenPackageParams(undefined)).toEqual({});
+    });
+  });
+
+  describe('when there is a single cookie header', () => {
+    it('returns an empty parameter set when our cookie is not there', () => {
+      const cookieHeader = '_st_fruit=banana';
+
+      expect(getOAuthTokenPackageParams(cookieHeader)).toEqual({});
+    });
+
+    it('returns the token package when our cookie is the only one', async () => {
+      const cookieHeader = `${tokenPackageCookie}`;
+
+      expect(getOAuthTokenPackageParams(cookieHeader)).toEqual(tokenPackageParams);
+    });
+
+    it('returns the token package when there are other cookies in the header', () => {
+      const cookieHeader = `_chocolate=chip; ${tokenPackageCookie}; _oatmeal=raisin`;
```

Comment:
This confuses me a bit as an example, because it reads to me as multiple cookies - if there isn't something I'm missing about that re: cookie behavior, I might suggest moving this particular test to the below `when there are multiple cookie headers` block

Reply:
A single cookie header [can contain multiple cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie#syntax), and a request can contain multiple cookie headers. This is an example of a single cookie header that contains multiple cookies.

Reply:
Right yeah, that makes sense, I think mentally I just group them all conceptually into the same block of "multiple cookies" 🤔 no worries though, it's a pretty minor thing and I should have clarified not a blocker/change request